streams/src/main/java/org/apache/kafka/streams/state/internals/MeteredKeyValueStore.java

```diff
@@ -182,7 +186,27 @@ public V delete(K key) {
 
     @Override
     public void flush() {
-        metrics.measureLatencyNs(time, flushDelegate, this.flushTime);
+        if (flushTime.shouldRecord()) {
+            measureLatency(new Action<V>() {
+                @Override
+                public V execute() {
+                    inner.flush();
+                    return null;
+                }
+            }, flushTime);
+        } else {
+            inner.flush();
+        }
+
+    }
+
+    private V measureLatency(final Action<V> action, final Sensor sensor) {
```

Comment:
Could you create a JIRA to keep track of this task: after java7 is dropped, we can add them as static functions for the `StreamsMetrics` interface itself (this is only supported in java8), and then also fix `ProcessorNode` and `StreamTask` to get rid of hard cast of `impl` classes.